src/cr/cube/utils/table_helper.py

```diff
@@ -0,0 +1,124 @@
+'''Grouping of various cube methods'''
+
+import numpy as np
+
+from ..dimension import Dimension
+
+
+class TableHelper(object):
+    '''Groups together useful cube utility methods.'''
```

Comment:
are tables part of a cube?  If this is a case, call it what it is, a DataTable or CubeTable and drop the Helper.  Also, perhaps create this lazily in Cube by attaching a cached property to the Cube object named table.

PS. I loath the term "Helper" for things, it's entirely non-descriptive.

Reply:
Addressed the nomenclature issues. Refactored a bit more. Also extracted `Means` and `Index` functionality into service classes.